src/crimson/osd/recovery_backend.cc

```diff
@@ -265,7 +265,6 @@ RecoveryBackend::scan_for_backfill(
       BackfillInterval bi;
       bi.begin = std::move(start);
       bi.end = std::move(next);
-      bi.version = pg.get_info().last_update;
```

Comment:
Do we still need this change if we will support projected log? I would prefer updating the interval version when we actually populate it so that the version will make sense.

btw, I've pushed a refactor to this code since it was quite confusing and there was a bug hidden there - https://github.com/ceph/ceph/pull/60522

Reply:
Yes, I think we still need this, otherwise, backfill interval's version may be ser to a relative new value, and skip some new modifications.

Reply:
I'm not sure I agree with this change (or perhaps not understanding it) , `version` is specifically there to know the version in which the scan has **occurred**.  

From the earlier discussion with @athanatos:
>Right, you want to capture last_update at the the time of the start of the scan because any update <= last_update is reflected in the local store (note, the primary and each replica tag their own interval with their own local last_update -- they won't necessarily match)

With the change here we will be capturing `last_update` after completing the scan while reacting to `PrimaryScanned`. Should we instead capture the version **before** the scan, in `PrimaryScanning::PrimaryScanning(my_context ctx)`?
Which as I can see we already are. We seem to overwrite the version  captured here (in `scan_for_backfill`) and this may have caused the issue?

```
// -- PrimaryScanning
BackfillState::PrimaryScanning::PrimaryScanning(my_context ctx)
  : my_base(ctx)
{
  backfill_state().backfill_info.version = peering_state().get_last_update(); <---
  backfill_listener().request_primary_scan(
    backfill_state().backfill_info.begin);
}
```

Seems like classic does the same thing by setting the version **before** scanning:
```
    bi->version = info.last_update;
    scan_range(local_min, local_max, bi, handle);
```

Reply:
> With the change here we will be capturing `last_update` after completing the scan while reacting to `PrimaryScanned`. Should we instead capture the version **before** the scan, in `PrimaryScanning::PrimaryScanning(my_context ctx)`

`PrimaryScanning::PrimaryScanning(my_context ctx)` already captures `peering_state.info.last_update`, see the following code: https://github.com/ceph/ceph/blob/de3c6e4ea1488028dfb66c20c23cb938b23cac88/src/crimson/osd/backfill_state.cc#L409C1-L409C78

Reply:
Ack, I'm planning to refactor a bit the way we handle interval scans and versions updates to make it easier to follow.